parse.lisp

```diff
@@ -32,19 +49,111 @@
 (defvar *parse-object-as-alist* nil
   "DEPRECATED, provided for backward compatibility")
 
+
 (defun make-adjustable-string ()
   "Return an adjustable empty string, usable as a buffer for parsing strings and numbers."
   (make-array +default-string-length+
               :adjustable t :fill-pointer 0 :element-type 'character))
 
+
+
+
+;; verify that the buffer contains a float, and is not a symbol like
```

Comment:
I'm not sure I like this code-duplication - first checking for validity, and then parsing, with a lot of code overlap.

That's why I asked about using `PARSE-FLOAT` - use an `IGNORE-ERRORS` around it to check or so, I hope this could be done in a single pass.

Let's see what CSR comes up with.